models/Asset.php

```diff
@@ -236,13 +236,8 @@ public static function getByPath($path, $force = false)
     protected static function typeMatch(Asset $asset)
     {
         $staticType = get_called_class();
-        if ($staticType != Asset::class) {
-            if (!$asset instanceof $staticType) {
-                return false;
-            }
-        }
 
-        return true;
+        return ($staticType === Asset::class) || ($asset instanceof $staticType) || is_subclass_of($staticType, Asset::class);
```

Comment:
what about situation when user requests wrong element from asset class e.g. `$asset = Pimcore\Model\Asset\Image::getById(1); // ID 1 is of folder type` introducing `is_subclass_of($staticType, Asset::class)` will return mismatched element